src/validators/frozenset.rs

```diff
@@ -33,16 +33,45 @@ impl Validator for FrozenSetValidator {
     ) -> ValResult<'data, PyObject> {
         let seq = input.validate_frozenset(extra.strict.unwrap_or(self.strict))?;
 
-        let length = seq.check_len(self.size_range, input)?;
+        let length = Some(seq.generic_len());
 
         let output = match self.item_validator {
             Some(ref v) => seq.validate_to_vec(py, length, v, extra, slots, recursion_guard)?,
             None => match seq {
-                GenericCollection::FrozenSet(f_set) => return Ok(f_set.into_py(py)),
+                GenericCollection::FrozenSet(set) => {
+                    seq.check_len(self.size_range, input)?;
+                    return Ok(set.into_py(py));
+                }
                 _ => seq.to_vec(py),
             },
         };
-        Ok(PyFrozenSet::new(py, &output)?.into_py(py))
+        let final_set = PyFrozenSet::new(py, &output)?;
+        if let Some((min_length, max_length)) = self.size_range {
```

Comment:
since all this is duplicated on set and frozen set, it might be worth moving it into a new function.

Reply:
I absolutely would love to crunch this part down, it was just more of a brute force for me. I couldn't for the life of me figure out how to feed final_set as an input for the check_len function as the types don't match up and I couldn't figure out how to convert it. Do you think it would be better to create a new function that close to check_len but with different params? I saw PrettyWood's solution was the expand parameters of check_len and let it accept a length input, which certainly seems to work. Or is there a straight forward way for me to feed final_set to check_len as the already existing input param?

Reply:
Honestly, probably better to create a new method, you could create a function to use in all scenarios, but it would require passing an anonymous function, or converting it to a macro, and it's really not worth it.

Reply:
you would need a trait, but better to implement it as a plain function which takes the vec as an argument.

It shouldn't touch the set/frozenset since that should be created after this check. See my other comment.

Reply:
I tried to get the method going for a while but im stuck on how to run the function prior to creating the set. I'm trying to figure out how i can get the set length from the vec, as they may be different lengths (i.e [1,2,3,4,4] would have 5 vec length but 4 set length). I tried running sort() and dedup() on the vec as well as turning it into a hashset to get its unique length, but neither worked as its a vector of PyAny. I could obviously create a PySet in the function and check its length, but I feel like thats likely defeating the purpose considering you said to check the length prior to creating the set.

Reply:
:facepalm: oh, good point, you're exactly right, I hadn't thought about that. This is me going too fast and not thinking straight, sorry about that.

At this point, I think I might need to do some experimentation. This consideration, together with the problems identified in #275 make the problem much more complex than I had realised.

Reply:
Oh, no worries, makes me feel a bit better that I wasn't just missing a super obvious solution.